src/Nethermind/Nethermind.Optimism/OptimismBaseFeeCalculator.cs

```diff
@@ -23,7 +25,23 @@ public UInt256 Calculate(BlockHeader parent, IEip1559Spec specFor1559)
         if (parent.Timestamp >= holoceneTimestamp)
         {
             // NOTE: This operation should never fail since headers should be valid at this point.
-            if (!parent.TryDecodeEIP1559Parameters(out EIP1559Parameters eip1559Params, out var error))
+            if (!HoloceneExtraParams.TryParse(parent, out HoloceneExtraParams parameters, out var error))
+            {
+                throw new InvalidOperationException($"{nameof(BlockHeader)} was not properly validated: {error}");
+            }
+
+            spec = new OverridableEip1559Spec(specFor1559)
+            {
+                ElasticityMultiplier = parameters.Elasticity,
+                BaseFeeMaxChangeDenominator = parameters.Denominator
+            };
+        }
+
+        if (false /* parent.Timestamp >= jovianTimestamp */)
+#pragma warning disable CS0162 // Unreachable code detected
+        {
+            // NOTE: This operation should never fail since headers should be valid at this point.
+            if (!JovianExtraParams.TryParse(parent, out JovianExtraParams eip1559Params, out var error))
```

Comment:
This check is quite similar to the one above. There is room for improvement to avoid duplication.